### PR TITLE
feat(contest/common-mistakes.md): merge undefined behavior list from #4472

### PR DESCRIPTION
**Pull request by inclyc**
Cherry-pick some undefined behaviors from OI-wiki/OI-wiki#4472.

Link: https://github.com/OI-wiki/OI-wiki/pull/4472

### CODE REVIEW

**Review by Xeonacid — DISMISSED** [review]

File: docs/contest/common-mistakes.md [hunk 1]
```diff
@@ -311,6 +311,38 @@ author: H-J-Granger, orzAtalod, ksyx, Ir1d, Chrogeek, Enter-tainer, yiyangit, sh
...
+    
+    可能被编译器直接优化为：
+    
+    ```cpp
+    int foo(int x) {
+        return 0;
+    }
+    ```
+    
+    因为编译器可以假定有符号整数永远不会溢出，因此 `x > x + 1` 恒成立，。
```
Comment (Tiphereth-A) [inline docs/contest/common-mistakes.md:333]:
```suggestion
    因为编译器可以假定有符号整数永远不会溢出，因此 `x > x + 1` 恒成立。
```

**Comment by inclyc** [comment]
Nice catch! Thanks @Tiphereth-A

**Review by Xeonacid — DISMISSED** [review]

**Comment by Enter-tainer** [comment]
@24OI-bot lint pls

**Comment by Ir1d** [comment]
@24OI-bot lint please

**Review by Enter-tainer — APPROVED** [review]
looks great to @24OI-bot

**Review by Backl1ght — APPROVED** [review]

**Pull request merged**